In [1]:
!pwd

/content


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [4]:
!ls

sample_data


Linear regression done on the Boston housing dataset

In [5]:
from keras.datasets import boston_housing

(x_train, y_train), (x_test, y_test) = boston_housing.load_data()

57344/57026 [==============================] - 0s 0us/step


In [6]:
print(f'Training data : {x_train.shape}')
print(f'Test data : {x_test.shape}')
print(f'Training sample : {x_train[0]}')
print(f'Training target sample : {x_train[0]}')

Training data : (404, 13)
Test data : (102, 13)
Training sample : [  1.23247   0.        8.14      0.        0.538     6.142    91.7
   3.9769    4.      307.       21.      396.9      18.72   ]
Training target sample : [  1.23247   0.        8.14      0.        0.538     6.142    91.7
   3.9769    4.      307.       21.      396.9      18.72   ]


In [7]:
mean = x_train.mean(axis=0)
x_train -= mean
std = x_train.std(axis=0)
x_train /= std

x_test -= mean
x_test /= std

In [8]:
from keras.models import Sequential
from keras.layers import Dense
from keras.wrappers.scikit_learn import KerasRegressor
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold

In [23]:
# create model
model = Sequential()
model.add(Dense(10, input_dim=13, kernel_initializer='normal', activation='relu'))
model.add(Dense(1, kernel_initializer='normal'))
# Compile model
model.compile(loss='mean_squared_error', optimizer='adam')

In [24]:
model.fit(x_train,y_train,epochs= 180)

Epoch 1/180
13/13 [==============================] - 0s 1ms/step - loss: 597.5584
Epoch 2/180
13/13 [==============================] - 0s 1ms/step - loss: 604.5415
Epoch 3/180
13/13 [==============================] - 0s 1ms/step - loss: 601.3013
Epoch 4/180
13/13 [==============================] - 0s 1ms/step - loss: 569.3563
Epoch 5/180
13/13 [==============================] - 0s 1ms/step - loss: 584.8391
Epoch 6/180
13/13 [==============================] - 0s 1ms/step - loss: 577.6221
Epoch 7/180
13/13 [==============================] - 0s 1ms/step - loss: 561.3925
Epoch 8/180
13/13 [==============================] - 0s 1ms/step - loss: 591.2436
Epoch 9/180
13/13 [==============================] - 0s 1ms/step - loss: 539.4092
Epoch 10/180
13/13 [==============================] - 0s 1ms/step - loss: 541.3753
Epoch 11/180
13/13 [==============================] - 0s 1ms/step - loss: 538.8785
Epoch 12/180
13/13 [==============================] - 0s 1ms/step - loss: 503.4856
Epoch 13/180


In [25]:
model.evaluate(x_test,y_test)

4/4 [==============================] - 0s 2ms/step - loss: 22.8735


22.873546600341797

Got accuracy of 22.87 on Boston dataset
Score of the model differ everytime as we train it. Because it is fed with different point of data using cross_validataion function. Like in the first runtime it's fed with 1-10 indexs in the 100 dataset next time it's fed with 10-20, 20-30....80-90...etc,. So the fit of the line changes as the accuracy with epoch of 180 and activation function is relu and 13 input parameters, with 10 neurons in the FIRST hidden layer specified in create model cell#(model.add(Dense(10, input_dim=13, kernel_initializer='normal', activation='relu'))#)